apollo-federation/src/connectors/runtime/errors.rs

```diff
@@ -0,0 +1,103 @@
+use serde_json_bytes::ByteString;
+use serde_json_bytes::Map;
+use serde_json_bytes::Value;
+
+use crate::connectors::Connector;
+use crate::connectors::runtime::key::ResponseKey;
+
+#[derive(Debug)]
+pub struct RuntimeError {
+    pub message: String,
+    code: Option<String>,
+    pub coordinate: Option<String>,
+    pub subgraph_name: Option<String>,
+    pub response_key: ResponseKey,
+    pub extensions: Map<ByteString, Value>,
+}
+
+impl RuntimeError {
+    pub fn new(message: impl Into<String>, response_key: &ResponseKey) -> Self {
```

Comment:
I think it is better to be explicit when cloning than passing a reference to clone hiddenly.

Reply:
no more cloning!